Assets/MixedRealityToolkit-SDK/Features/Teleportation/MixedRealityTeleportManager.cs

```diff
@@ -0,0 +1,263 @@
+
+#if UNITY_EDITOR
+            if (!UnityEditor.EditorApplication.isPlaying)
+            {
+
+                var eventSystems = Object.FindObjectsOfType<EventSystem>();
+
+                if (eventSystems.Length == 0)
+                {
+                    if (!MixedRealityManager.Instance.ActiveProfile.IsInputSystemEnabled)
+                    {
+                        new GameObject("Event System").AddComponent<EventSystem>();
+                    }
+                    else
+                    {
+                        Debug.Log("The Input System didn't properly add an event system to your scene. Please make sure the Input System's priority is set higher than the teleport system.");
+                    }
+                }
+                else if (eventSystems.Length > 1)
+                {
+                    Debug.Log("Too many event systems in the scene. The Teleport System requires only one.");
+                }
+            }
+#endif // UNITY_EDITOR
+
+            TeleportDuration = MixedRealityManager.Instance.ActiveProfile.TeleportDuration;
+            teleportEventData = new TeleportEventData(EventSystem.current);
+        }
+
+        #endregion IMixedRealityManager Implementation
+
+        #region IEventSystemManager Implementation
+
+        /// <inheritdoc />
+        public override void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.EventFunction<T> eventHandler)
+        {
+            Debug.Assert(eventData != null);
+            var teleportData = ExecuteEvents.ValidateEventData<TeleportEventData>(eventData);
+            Debug.Assert(teleportData != null);
+            Debug.Assert(!teleportData.used);
+
+            // Process all the event listeners
+            base.HandleEvent(teleportData, eventHandler);
+        }
+
+        /// <summary>
+        /// Unregister a <see cref="GameObject"/> from listening to Teleport events.
+        /// </summary>
+        /// <param name="listener"></param>
+        public override void Register(GameObject listener)
+        {
+            base.Register(listener);
+        }
+
+        /// <summary>
+        /// Unregister a <see cref="GameObject"/> from listening to Teleport events.
+        /// </summary>
+        /// <param name="listener"></param>
+        public override void Unregister(GameObject listener)
+        {
+            base.Unregister(listener);
+        }
+
+        #endregion IEventSystemManager Implementation
+
+        #region IMixedRealityTeleportSystem Implementation
+
+        public float TeleportDuration
+        {
+            get { return teleportDuration; }
+            set
+            {
+                if (isProcessingTeleportRequest)
+                {
+                    Debug.LogWarning("Couldn't change teleport duration. Teleport in progress.");
+                    return;
+                }
+
+                teleportDuration = value;
+            }
+        }
+
+        private float teleportDuration = 0.25f;
```

Comment:
nit: define above the property (better matches the rest of the code base)

Reply:
I don't understand.

Reply:
other files in our project define backing fields before the property. ex:

private bool exampleField = false;
public bool ExampleProperty
{ … }

Reply:
private field before property. Got it.